contrib/mpi-proxy-split/mpi-wrappers/mana_p2p_update_logs.c

```diff
@@ -68,11 +70,14 @@ void fill_in_log(struct p2p_log_msg *p2p_log) {
     }
   }
   close(fd2);
+#ifdef 0
+  // The following code has infinite loop. Comment out for now.
   fd2 = dup(fd_request);
   while (1) {
     readall(fd2, &p2p_request, sizeof(p2p_request));
     if (p2p_request.request == MPI_REQUEST_NULL) {
       readall(fd_request, &p2p_request, sizeof(p2p_request));
```

Comment:
@gc00 The line 63-69 already find the same request matching the given p2p_log->request. Why do we need this loop from line 76-80. If we break when readall() return 0, then most likely we will keep reading until the end of the file. However, we did not use the read buffer p2p_request anymore after we break from this loop. Should we simply delete the loop of line 75 to line 80?

Reply:
@dahongli, You're right.  When I was writing this code, I was assuming that we would log for a long time, but only replay for a short time (until we reach the issue related to a bug).  So, I never thought about what happens if we are replaying, and we reach the end of the log.

I'm not sure right now what is the right thing to do in this case.  We could just print a message saying that we have reached the end of the log, and then unset the environment variable `MANA_P2P_REPLAY`.

I'm not sure what is the correct behavior.  If you like the behavior that I suggest, please implement it.  Or feel free to propose a different behavior.

Reply:
This code updates log messages p2p_log_<rank>.txt bases on the receiving order of the requests from p2p_log_request_<rank>.txt. This update should happen before we replay the log messages in checkpointing restart. The general workflow I think something like
1. set `MANA_P2P_LOG`
2. launch application. This would create both message log file p2p_log_<rank>.txt and request log file p2p_log_request_<rank>.txt
3. Checkpointing
4. Let the application keep running until the application dies or stops
5. Now we run mana_p2p_update_log. At this moment, we have not restart from checkpointing image yet. 
6. Set replay environment variable `MANA_P2P_REPALY`
7. Restart checkpointing 

If we do the above workflow, this mana_p2p_update_log code only needs to read a request from request log and find the matching request in the message log and do the update. We do not need the loop from line 73 to 80. We could just delete them for now. 

The drawback is the log file records the sequence of all messages and receiving requests. It is not necessary. In the future, we could optimize it by dynamically enable or disable the logging of a running processes. We could this optimization later.